File: practice/D_Small_GCD.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>
#include <cmath>

int solve(std::vector<int>& nums, int n) {
    int cnt = 0;
    for (int i = 0; i < n; ++i) {
        for (int j = i + 1; j < n; ++j) {
            int gcd = std::__gcd(nums[i], nums[j]);
            cnt += gcd * (n - j - 1);
        }
    }
    return cnt;
}

int main() {
    int t;
    std::cin >> t;

    while (t--) {
        int n;
        std::cin >> n;

        std::vector<int> nums(n);
        for (int i = 0; i < n; ++i) {
            std::cin >> nums[i];
        }

        std::sort(nums.begin(), nums.end());

        int ans = solve(nums, n);
        std::cout << ans << std::endl;
    }

    return 0;
}

File: practice/D_Small_GCD.py
import math


def solve(nums, n):
    cnt = 0
    for i in range(n):
        for j in range(i + 1, n):
            gcd = math.gcd(nums[i], nums[j])
            cnt += gcd * (n - j - 1)
            
    return cnt


def main():
    t = int(input())

    for _ in range(t):
        n = int(input())
        nums = list(map(int, input().split()))
        nums.sort()

        ans = solve(nums, n)
        print(ans)


if __name__ == "__main__":
    main()
